theano/gpuarray/magma_inv.c

```diff
@@ -0,0 +1,101 @@
+#section init_code
+
+setup_ext_cuda();
+
+#section support_code_struct
+
+int APPLY_SPECIFIC(magma_inv)(PyGpuArrayObject *A, PyGpuArrayObject **_A_inv,
+                              PyGpuContextObject *c) {
+  PyGpuArrayObject *A_inv = *_A_inv;
```

Comment:
I think you should always use `*_A_inv` instead of a temporar value `A_inv`. If a copy of `A_inv` is done (at line 47) but an error occurs after that and before setting `*_A_inv = A_inv`, then the memory allocated for the copy of `A_inv` will not be freed (as it is not checked at all at `fail` label) and will not be seen/handled by Theano (as `*_A_inv` will have not been updated).